packages/rest-crud/src/crud-rest.controller.ts

```diff
@@ -0,0 +1,320 @@
+ * @param modelCtor A model class, e.g. `Product`.
+ * @param options Configuration options, e.g. `{basePath: '/products'}`.
+ */
+export function defineCrudRestController<
+  T extends Entity,
+  IdType,
+  IdName extends keyof T,
+  Relations extends object = {}
+>(
+  modelCtor: typeof Entity & {prototype: T & {[key in IdName]: IdType}},
+  options: CrudRestControllerOptions,
+): CrudRestControllerCtor<T, IdType, IdName, Relations> {
+  const modelName = modelCtor.name;
+  const idPathParam: ParameterObject = {
+    name: 'id',
+    in: 'path',
+    schema: getIdSchema(modelCtor),
+  };
+
+  @api({basePath: options.basePath, paths: {}})
+  class CrudRestControllerImpl
+    implements CrudRestController<T, IdType, IdName> {
+    constructor(
+      public readonly repository: EntityCrudRepository<T, IdType, Relations>,
+    ) {}
+
+    @post('/', {
+      ...response.model(200, `${modelName} instance created`, modelCtor),
+    })
+    async create(
+      @body(modelCtor, {exclude: modelCtor.getIdProperties() as (keyof T)[]})
+      data: Omit<T, IdName>,
+    ): Promise<T> {
+      return this.repository.create(
+        // FIXME(bajtos) Improve repository API to support this use case
+        // with no explicit type-casts required
+        data as DataObject<T>,
+      );
+    }
+
+    @get('/', {
+      ...response.array(200, `Array of ${modelName} instances`, modelCtor, {
+        includeRelations: true,
+      }),
+    })
+    async find(
+      @param.query.object('filter', getFilterSchemaFor(modelCtor))
+      filter?: Filter<T>,
+    ): Promise<(T & Relations)[]> {
+      return this.repository.find(filter);
+    }
+
+    @get('/{id}', {
+      ...response.model(200, `${modelName} instance`, modelCtor, {
+        includeRelations: true,
+      }),
+    })
+    async findById(
+      @param(idPathParam) id: IdType,
+      @param.query.object('filter', getFilterSchemaFor(modelCtor))
+      filter?: Filter<T>,
+    ): Promise<T & Relations> {
+      return this.repository.findById(id, filter);
+    }
+
+    @get('/count', {
+      ...response(200, `${modelName} count`, {schema: CountSchema}),
+    })
+    async count(
+      @param.query.object('where', getWhereSchemaFor(modelCtor))
+      where?: Where<T>,
+    ): Promise<Count> {
+      return this.repository.count(where);
+    }
+
+    @patch('/', {
+      ...response(200, `Count of ${modelName} models updated`, {
+        schema: CountSchema,
+      }),
+    })
+    async updateAll(
+      @body(modelCtor, {partial: true}) data: Partial<T>,
+      @param.query.object('where', getWhereSchemaFor(modelCtor))
+      where?: Where<T>,
+    ): Promise<Count> {
+      return this.repository.updateAll(
+        // FIXME(bajtos) Improve repository API to support this use case
+        // with no explicit type-casts required
+        data as DataObject<T>,
+        where,
+      );
+    }
+
+    @patch('/{id}', {
+      responses: {
+        '204': {description: `${modelName} was updated`},
```

Comment:
`204` should be `200`

Reply:
`PATCH /my-model/{id}` does not return any response data, that's why the status code is 204. See also the controller template that uses 204 too.

https://github.com/strongloop/loopback-next/blob/554c763aa5c8db92bb4c3e5549960b8a99673791/packages/cli/generators/controller/templates/src/controllers/controller-rest-template.ts.ejs#L116-L122